capstone/src/main/resources/static/user.js

```diff
@@ -393,26 +357,32 @@ function createCalendar() {
   });
 }
 
-/* helper function to create calendar event elements */
-function createCalendarEventElement(event, eventTime) {
-  const eventDisplay = document.createElement("div");
-  eventDisplay.setAttribute("class", "event general-container col");
-  const pElementTitle = document.createElement('p');
-  pElementTitle.innerText = event.eventTitle;
-  eventDisplay.appendChild(pElementTitle);
-  const pElementTime = document.createElement('p');
-  pElementTime.innerText = eventTime.toDateString();
-  eventDisplay.appendChild(pElementTime);
-  return eventDisplay;
+/* helper function to create calendar events */
+function createCalendarEvent(event, today, endDate, borderColor) {
+  var eventDate = new Date(event.eventDateTime);
+  if (eventDate.getTime() > today.getTime() && eventDate.getTime() < endDate.getTime()) {
+    var diff = eventDate.getDate() - today.getDate();
+    const generalDateDiv = document.getElementById("date" + diff);
+    const newEvent = createEventElement(generalDateDiv, event, false, true, false);
+    newEvent.style.borderColor = borderColor;
+  }
 }
-
 /* function to create a public profile of an organization */
 function getPublicProfile() {
- var organizationId = window.location.hash.substring(1);
- fetch('get-public-profile?organization-id=' + organizationId).then(response => response.json()).then((data) => {
-   createProfile(data, false, true);
-   const eventDiv = document.getElementById("hosted-events");
-   data.events.forEach((event) => eventDiv.appendChild(createSavedEventElement(event, true, false)));
- });
+  fetch('user-info').then(response => response.json()).then((data) => {
+     if (data.userType != "unknown") {
+       var organizationId = window.location.hash.substring(1);
```

Comment:
If you use the query string (`?key=value`), you can use URLSearchParams to extract individual values.

https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams